File: presentation/mlx_tts/src/data/voices.ts
export interface Voice {
  id: string;
  name: string;
  description: string;
  lang: string;
  gender: "male" | "female";
}

export interface VoiceGroup {
  language: string;
  prefix: string;
  color: string;
  voices: Voice[];
}

export const voiceGroups: VoiceGroup[] = [
  {
    language: "American English",
    prefix: "a",
    color: "#60a5fa",
    voices: [
      { id: "af_heart", name: "Heart", description: "Warm & expressive", lang: "en-us", gender: "female" },
      { id: "af_sarah", name: "Sarah", description: "Clear & professional", lang: "en-us", gender: "female" },
      { id: "af_bella", name: "Bella", description: "Bright & energetic", lang: "en-us", gender: "female" },
      { id: "af_sky", name: "Sky", description: "Calm & soothing", lang: "en-us", gender: "female" },
      { id: "af_nicole", name: "Nicole", description: "Natural conversational", lang: "en-us", gender: "female" },
      { id: "af_nova", name: "Nova", description: "Smooth & confident", lang: "en-us", gender: "female" },
      { id: "am_adam", name: "Adam", description: "Deep & authoritative", lang: "en-us", gender: "male" },
      { id: "am_michael", name: "Michael", description: "Friendly & clear", lang: "en-us", gender: "male" },
      { id: "am_echo", name: "Echo", description: "Resonant", lang: "en-us", gender: "male" },
      { id: "am_liam", name: "Liam", description: "Casual & relaxed", lang: "en-us", gender: "male" },
    ],
  },
  {
    language: "British English",
    prefix: "b",
    color: "#a78bfa",
    voices: [
      { id: "bf_emma", name: "Emma", description: "Elegant & precise", lang: "en-gb", gender: "female" },
      { id: "bf_isabella", name: "Isabella", description: "Warm & refined", lang: "en-gb", gender: "female" },
      { id: "bm_george", name: "George", description: "Classic & distinguished", lang: "en-gb", gender: "male" },
      { id: "bm_lewis", name: "Lewis", description: "Calm & measured", lang: "en-gb", gender: "male" },
      { id: "bm_daniel", name: "Daniel", description: "Formal newsreader", lang: "en-gb", gender: "male" },
    ],
  },
  {
    language: "Mandarin Chinese",
    prefix: "z",
    color: "#f87171",
    voices: [
      { id: "zf_xiaobei", name: "小北", description: "Lively & youthful", lang: "zh", gender: "female" },
      { id: "zf_xiaoni", name: "小妮", description: "Gentle & warm", lang: "zh", gender: "female" },
      { id: "zm_yunjian", name: "云健", description: "Deep broadcast voice", lang: "zh", gender: "male" },
      { id: "zm_yunxi", name: "云希", description: "Natural & conversational", lang: "zh", gender: "male" },
    ],
  },
  {
    language: "Japanese",
    prefix: "j",
    color: "#4ade80",
    voices: [
      { id: "jf_alpha", name: "Alpha", description: "Clear & expressive", lang: "ja", gender: "female" },
      { id: "jf_gongitsune", name: "Gongitsune", description: "Storyteller style", lang: "ja", gender: "female" },
      { id: "jm_kumo", name: "Kumo", description: "Calm & measured", lang: "ja", gender: "male" },
    ],
  },
];

File: mlx_tts/voices.py
"""
Kokoro voice catalog.

Prefix legend:
  a = American, b = British
  f = female,   m = male
"""

KOKORO_VOICES: dict[str, dict] = {
    # ---------- American Female ----------
    "af_heart":    {"lang": "en-us", "gender": "female", "accent": "american", "note": "warm & expressive (default)"},
    "af_sarah":    {"lang": "en-us", "gender": "female", "accent": "american", "note": "clear & professional"},
    "af_bella":    {"lang": "en-us", "gender": "female", "accent": "american", "note": "bright & energetic"},
    "af_sky":      {"lang": "en-us", "gender": "female", "accent": "american", "note": "calm & soothing"},
    "af_nicole":   {"lang": "en-us", "gender": "female", "accent": "american", "note": "natural conversational"},
    "af_nova":     {"lang": "en-us", "gender": "female", "accent": "american", "note": "smooth & confident"},
    # ---------- American Male ------------
    "am_adam":     {"lang": "en-us", "gender": "male",   "accent": "american", "note": "deep & authoritative"},
    "am_michael":  {"lang": "en-us", "gender": "male",   "accent": "american", "note": "friendly & clear"},
    "am_echo":     {"lang": "en-us", "gender": "male",   "accent": "american", "note": "resonant"},
    "am_liam":     {"lang": "en-us", "gender": "male",   "accent": "american", "note": "casual & relaxed"},
    # ---------- British Female -----------
    "bf_emma":     {"lang": "en-gb", "gender": "female", "accent": "british",  "note": "elegant & precise"},
    "bf_isabella": {"lang": "en-gb", "gender": "female", "accent": "british",  "note": "warm & refined"},
    # ---------- British Male -------------
    "bm_george":   {"lang": "en-gb", "gender": "male",   "accent": "british",  "note": "classic & distinguished"},
    "bm_lewis":    {"lang": "en-gb", "gender": "male",   "accent": "british",  "note": "calm & measured"},
    "bm_daniel":   {"lang": "en-gb", "gender": "male",   "accent": "british",  "note": "formal newsreader style"},
}

DEFAULT_VOICE = "af_heart"


def list_voices(gender: str = None, accent: str = None) -> dict[str, dict]:
    """Return filtered voice catalog."""
    result = KOKORO_VOICES
    if gender:
        result = {k: v for k, v in result.items() if v["gender"] == gender.lower()}
    if accent:
        result = {k: v for k, v in result.items() if v["accent"] == accent.lower()}
    return result


def voice_info(voice_id: str) -> dict | None:
    return KOKORO_VOICES.get(voice_id)
